openedx_lti_tool_plugin/models.py

```diff
@@ -82,33 +84,84 @@ def __init__(self, *args: tuple, **kwargs: dict):
         self._initial_pii = self.pii
 
     @property
-    def username(self) -> str:
-        """str: Username."""
-        return f'{app_config.name}.{self.uuid}'
+    def short_uuid(self) -> str:
+        """str: Short UUID."""
+        return shortuuid.encode(self.uuid)[:8]
 
     @property
-    def email(self) -> str:
-        """str: Email address."""
-        return f'{self.uuid}@{app_config.name}'
+    def given_name(self) -> str:
+        """str: Given name."""
+        return self.pii.get('given_name', '')
+
+    @property
+    def middle_name(self) -> str:
+        """str: Middle name."""
+        return self.pii.get('middle_name', '')
+
+    @property
+    def family_name(self) -> str:
+        """str: Family name."""
+        return self.pii.get('family_name', '')
 
     @property
     def name(self) -> str:
         """str: Name."""
-        name = self.pii.get('name', '')
-        given_name = self.pii.get('given_name', '')
-        middle_name = self.pii.get('middle_name', '')
-        family_name = self.pii.get('family_name', '')
+        if self.given_name and self.middle_name and self.family_name:
+            return f'{self.given_name} {self.middle_name} {self.family_name}'
+
+        if self.given_name and self.family_name:
+            return f'{self.given_name} {self.family_name}'
+
+        return self.given_name
+
+    @property
+    def username(self) -> str:
```

Comment:
Are you planning to perform a migration of the users currently existing? With this new property wouldn't be a discrepancy between the existing user models username and the username generated by this property?

Reply:
@alexjmpb There will be a discrepancy with the structure of the username of the older LtiProfiles with the new LtiProfiles. 

Right now there is no requirement to update the username of the older LtiProfiles, if this is required in the future we could execute a backfill to fix this discrepancy.